hedera-mirror-test/src/test/java/com/hedera/mirror/test/e2e/acceptance/steps/EstimatePrecompileFeature.java

```diff
@@ -754,17 +759,45 @@ public void createFungibleTokenEstimateGas() {
         ByteBuffer encodedFunctionCall = getFunctionFromEstimateArtifact(ContractMethods.CREATE_FUNGIBLE_TOKEN)
                 .encodeCallWithArgs(asAddress(admin.getAccountId().toSolidityAddress()));
 
+        try {
+            validateGasEstimationForCreateToken(
+                    Strings.encode(encodedFunctionCall),
+                    ContractMethods.CREATE_FUNGIBLE_TOKEN.getActualGas(),
+                    calculateCreateTokenFee(1, true));
+            return;
+
+        } catch (AssertionError e) {
+            log.warn("Assertion failed for estimateGas with current exchange rates. Trying with next exchange rates.");
+        } catch (Exception e) {
+            log.warn(
+                    "Exception occurred for estimateGas with current exchange rates. Trying with next exchange rates.");
+        }
         validateGasEstimationForCreateToken(
```

Comment:
Make this a lambda `Consumer<Boolean>` that takes in current flag then move try/catch retry into a helper method

```suggestion
        var estimateFunction = current -> validateGasEstimationForCreateToken(
                    Strings.encode(encodedFunctionCall),
                    ContractMethods.CREATE_FUNGIBLE_TOKEN.getActualGas(),
                    calculateCreateTokenFee(1, current));
        validateGasEstimation(estimateFunction);
```

Reply:
Implemented